src/AutoRest.CSharp/LowLevel/Generation/LowLevelExampleComposer.cs

```diff
@@ -0,0 +1,1040 @@
+            if (clientConstructor.Parameters.Any(p => p.Type.EqualsIgnoreNullable(KeyAuthType)))
+            {
+                builder.AppendLine("var credential = new AzureKeyCredential(\"<key>\");");
+            }
+            else if (clientConstructor.Parameters.Any(p => p.Type.EqualsIgnoreNullable(TokenAuthType)))
+            {
+                builder.AppendLine("var credential = new DefaultAzureCredential();");
+            }
+
+            if (clientConstructor.Parameters.Any(p => p.Type.EqualsIgnoreNullable(UriType)))
+            {
+                builder.AppendLine($"var endpoint = new Uri(\"<{GetEndpoint()}>\");");
+            }
+
+            var code = $"var client = new {invocationChain[0].Name}({MockClientConstructorParameterValues(invocationChain[0].Parameters)})";
+
+            if (invocationChain.Count > 1)
+            {
+                for (int i = 1; i < invocationChain.Count; i++)
+                {
+                    var factoryMethod = invocationChain[i];
+                    code += GetFactoryMethodCode(factoryMethod);
+                }
+            }
+            builder.Append(code).AppendLine(";");
+        }
+
+        private string MockClientConstructorParameterValues(IReadOnlyList<Parameter> parameters)
+        {
+            var parameterValues = new List<string>(parameters.Count);
+            foreach (var parameter in parameters)
+            {
+                if (parameter.Type.EqualsIgnoreNullable(UriType))
+                {
+                    parameterValues.Add("endpoint");
+                }
+                else if (parameter.Name.Equals("endpoint", StringComparison.OrdinalIgnoreCase))
+                {
+                    // sometimes the endpoint parameter cannot be generated as Uri type, best efforts to guesss it
+                    // see: https://github.com/Azure/autorest/issues/4571
+                    parameterValues.Add($"\"<{GetEndpoint()}>\"");
+                }
+                else if (parameter.Type.EqualsIgnoreNullable(KeyAuthType) || parameter.Type.EqualsIgnoreNullable(TokenAuthType))
+                {
+                    parameterValues.Add("credential");
+                }
+                else
+                {
+                    parameterValues.Add(MockParameterValue(parameter));
+                }
+            }
+            return string.Join(", ", parameterValues);
+        }
+
+        /// <summary>
+        /// Get the methods to be called to get the client, it should be like `Client(...).GetXXClient(..).GetYYClient(..)`.
+        /// It's composed of a constructor of non-subclient and a optional list of subclient factory methods.
+        /// </summary>
+        /// <returns></returns>
+        private IReadOnlyList<MethodSignatureBase> GetClientInvocationChain()
+        {
+            var chain = new List<MethodSignatureBase>();
+            foreach (var client in Context.Library.RestClients.Where(r => !r.IsSubClient))
```

Comment:
Here we are recovering information that is already available in when `LowLevelOutputLibraryFactory.CreateClients` is called. We should preserve this call chain and simply reference it here.

Reply:
－I changed to keep the client hierarchy paths info in `LowLevelOutputLibrary`.
－Keep the client invocation info as a property in `LowLevelExampleComposer`, because each instance of `LowLevelExampleComposer` is bound to one client.

Reply:
Well, this isn't what I meant. In fact, using `Lazy` makes the thing even worse.
When we create clients, we have all information about "parent client" - "subclient" relationship, so we can preserve it and then just iterate through the parents. I've created a PR that you can use: https://github.com/Azure/autorest.csharp/pull/2442